WeatherApp/WeatherApp.Webpage/Services/AutocompleteService.cs

```diff
@@ -1,30 +1,30 @@
 ﻿using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
 using Newtonsoft.Json.Linq;
 using System;
 using System.Collections.Generic;
-using System.Linq;
 using System.Net.Http;
-using System.Threading.Tasks;
 using WeatherApp.WebSite.Models;
 
 namespace WeatherApp.WebSite.Services
 {
-    public class AutocompleteService
+    public class AutocompleteService : IAutocompleteService
     {
-        const string API_KEY = "jzxwrjVpz590MChJ0KjlrLnwg_syikNAPYB0tvSemLE";
+        readonly string _apiKey;
+        readonly string _baseUrl;
 
-        public AutocompleteService(IWebHostEnvironment webHostEnvironment)
+        public AutocompleteService(IConfiguration configuration)
         {
-            WebHostEnvironment = webHostEnvironment;
+            _apiKey = configuration.GetValue<string>("ApiKeys:Autocomplete");
+            _baseUrl = configuration.GetValue<string>("ApiBaseUrls:Autocomplete");
         }
-        
-        public IWebHostEnvironment WebHostEnvironment { get; }
 
         public IEnumerable<Location> GetSuggestions(string query)
         {
-            string jsonString = "";
-            string url = $"https://autocomplete.geocoder.ls.hereapi.com/6.2/suggest.json?query={query}&maxresults=5&resultType=city&language=en&apikey={API_KEY}";
+            string urlParameters = $"apikey={_apiKey}&query={query}&maxresults=5&resultType=city&language=en";
```

Comment:
Be consistent with type information of your variables. Use explicit type everywhere or use `var` everywhere but not mix them.

Reply:
Thank you for your remark, it is fixed now.